File: Patterns/Strategy/java/src/PatternStrategy.java
import heroes.Hawkeye;
import heroes.Hero;
import heroes.ShillenKnight;
import heroes.SpellSinger;
import weapons.BowWeaponBehavior;


public class PatternStrategy {

	/**
	 * @param args
	 */
	public static void main(String[] args) {
		Hero magican = new SpellSinger();
		Hero archer = new Hawkeye();
		Hero tank = new ShillenKnight();
		
		tank.attack();
		tank.setWeapon(new BowWeaponBehavior());
		tank.attack();
		
		archer.attack();
		magican.attack();
	
	}

}

File: Patterns/Strategy/php/www/PatternStrategy.php
<?php
	require_once 'heroes/ShillenKnight.php';
	require_once 'heroes/Hawkeye.php';
	require_once 'heroes/SpellSinger.php';
	
	require_once 'weapons/BowWeaponBehavior.php';
	require_once 'weapons/HydroblastMagicBehavior.php';
	require_once 'weapons/SwordWeaponBehavior.php';
	
	class PatternStrategy{
		public function __construct(){
			$magican = new SpellSinger;
			$archer = new Hawkeye;
			$tank = new ShillenKnight;
			
			$tank->attack();
			$tank->setWeapon(new BowWeaponBehavior);
			$tank->attack();

			$archer->attack();
			$magican->attack();
		}
	}
